### PR TITLE
Introduces the Loadify Component

### PR DESCRIPTION
**Pull request by amir-ba**
### Description:
This Introduces a HOC of `LoadingComponent`.
When the component is set to be loading the Wrapped component will be shown with a loading spinner overlay.

Plz review.

### CODE REVIEW

File: src/HigherOrderComponent/LoadingComponent/LoadingComponent.example.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+This demonstrates the use of `LoadingComponent`.
```
Comment (marcjansen) [inline src/HigherOrderComponent/LoadingComponent/LoadingComponent.example.md]: of **the** `LoadingComponent`

File: src/HigherOrderComponent/LoadingComponent/LoadingComponent.example.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    return(
+      <div
+        id={this.mapDivId}
+        style={{
+          height: '400px'
+        }}
+      >
+        <LoadingTree 
+          spinning={this.state.loading}
+          tip={'Loading Text Here'}
```
[the remark this answers is not in the document]
Comment (marcjansen) [reply src/HigherOrderComponent/LoadingComponent/LoadingComponent.example.md]: :heart: for …

File: src/HigherOrderComponent/LoadingComponent/LoadingComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    static propTypes = {
+      /**
+       * whether it should be loading or not.
+       * For the List of all the Spin component properties see
+       * https://ant.design/components/spin/
+       * @type {Boolean}
+       */
+      spinning: PropTypes.bool,
+      /**
+       * Size of the loading elmenet.
```
Comment (marcjansen) [inline src/HigherOrderComponent/LoadingComponent/LoadingComponent.jsx]: typo: element

**Comment by amir-ba** [comment]
@marcjansen I have check the support for the existing loading and make it compatible thanks for reminding me.

**Comment by amir-ba** [comment]
@marcjansen regarding the existing `Loading` property of some antd components:
The proposed approach is useful when we want to wrap a component or components in a loader to show a loading sign. and while it doesn't interfere with the `loading` property of the antd components the existing antd property on some components shows the loader inside the component and our `Loadify`component wraps the component inside the loader. So I believe there are 2 different use cases.

File: src/HigherOrderComponent/Loadify/Loadify.jsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+ *
+ * Wrapped components will be used as the content of the Loader component.
+ * If the wrapped component is set to be loading the loader element
+ * will be shown and if not it wont.
+ *
+ * @param {Component} WrappedComponent The component to wrap and enhance.
+ * @param {Component} options The options to apply.
+ * @return {Component} The wrapped component.
+ */
+export function Loadify( WrappedComponent,{
```
Comment (dnlkoch) [inline src/HigherOrderComponent/Loadify/Loadify.jsx]: Minor, but please check usage of whitespaces here.

Comment (dnlkoch) [inline src/HigherOrderComponent/Loadify/Loadify.jsx]: `Loadify` is a method and should start with lower case.

File: src/HigherOrderComponent/Loadify/Loadify.spec.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    render() {
+      return (
+        <div>A mock Component</div>
+      );
+    }
+  }
+  /* eslint-enable require-jsdoc */
+
+  beforeEach(() => {
+    EnhancedComponent =  Loadify(MockComponent, {
```
Comment (dnlkoch) [inline src/HigherOrderComponent/Loadify/Loadify.spec.jsx]: Remove extra whitespace before `Loadify`.

File: src/HigherOrderComponent/Loadify/Loadify.spec.jsx [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+      });
+      const wrapperNoRef = TestUtil.mountComponent(EnhancedComponentNoRef, props);
+      const wrappedInstanceNoRef = wrapperNoRef.instance().getWrappedInstance();
+      expect(wrappedInstanceNoRef).toBeUndefined();
+    });
+
+    it('shows or hides the Loading component based on the spinning prop', () => {
+      let wrapper = TestUtil.mountComponent(EnhancedComponent);
+
+      // 1. Show loading sign.
```
Comment (dnlkoch) [inline src/HigherOrderComponent/Loadify/Loadify.spec.jsx]: Wrong comment, the wrapper is configured with `spinning: false`.

File: src/HigherOrderComponent/Loadify/Loadify.spec.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      wrapper = TestUtil.mountComponent(EnhancedComponent, {
+        spinning: false
+      });
+      expect(wrapper.find('.ant-spin').length).toBe(0);
+
+      // 2. Don't show loading sign.
```
Comment (dnlkoch) [inline src/HigherOrderComponent/Loadify/Loadify.spec.jsx]: Wrong comment, the wrapper is configured with `spinning: true`.

**Comment by dnlkoch** [comment]
Top! Please merge!

**Pull request merged**